daemon/libnetwork/internal/resolvconf/resolvconf.go

```diff
@@ -291,70 +289,108 @@ func (rc *ResolvConf) TransformForIntNS(
 // is true, the file will include header information if supplied, and a trailing
 // comment that describes how the file was constructed and lists external resolvers.
 func (rc *ResolvConf) Generate(comments bool) ([]byte, error) {
-	s := struct {
-		Md          *metadata
-		NameServers []netip.Addr
-		Search      []string
-		Options     []string
-		Other       []string
-		Overrides   []string
-		Comments    bool
-	}{
-		Md:          &rc.md,
-		NameServers: rc.nameServers,
-		Search:      rc.search,
-		Options:     rc.options,
-		Other:       rc.other,
-		Comments:    comments,
+	var b bytes.Buffer
+	b.Grow(512) // estimated size for a regular resolv.conf we produce.
+
+	if comments && rc.md.Header != "" {
+		b.WriteString(rc.md.Header + "\n")
+		b.WriteByte('\n')
+	}
+	for _, ns := range rc.nameServers {
+		b.WriteString("nameserver ")
+		b.WriteString(ns.String())
+		b.WriteByte('\n')
+	}
+	if len(rc.search) > 0 {
+		b.WriteString("search ")
+		for i, s := range rc.search {
+			if i > 0 {
+				b.WriteByte(' ')
+			}
+			b.WriteString(s)
+		}
+		b.WriteByte('\n')
 	}
-	if rc.md.NSOverride {
-		s.Overrides = append(s.Overrides, "nameservers")
-	}
-	if rc.md.SearchOverride {
-		s.Overrides = append(s.Overrides, "search")
-	}
-	if rc.md.OptionsOverride {
-		s.Overrides = append(s.Overrides, "options")
-	}
-
-	const templateText = `{{if .Comments}}{{with .Md.Header}}{{.}}
-
-{{end}}{{end}}{{range .NameServers -}}
-nameserver {{.}}
-{{end}}{{with .Search -}}
-search {{join . " "}}
-{{end}}{{with .Options -}}
-options {{join . " "}}
-{{end}}{{with .Other -}}
-{{join . "\n"}}
-{{end}}{{if .Comments}}
-# Based on host file: '{{.Md.SourcePath}}'{{with .Md.Transform}} ({{.}}){{end}}
-{{range .Md.Warnings -}}
-# {{.}}
-{{end -}}
-{{with .Md.ExtNameServers -}}
-# ExtServers: {{.}}
-{{end -}}
-{{with .Md.InvalidNSs -}}
-# Invalid nameservers: {{.}}
-{{end -}}
-# Overrides: {{.Overrides}}
-{{with .Md.NDotsFrom -}}
-# Option ndots from: {{.}}
-{{end -}}
-{{end -}}
-`
-
-	funcs := template.FuncMap{"join": strings.Join}
-	var buf bytes.Buffer
-	templ, err := template.New("summary").Funcs(funcs).Parse(templateText)
-	if err != nil {
-		return nil, errSystem{err}
+	if len(rc.options) > 0 {
+		b.WriteString("options ")
+		for i, s := range rc.options {
+			if i > 0 {
+				b.WriteByte(' ')
+			}
+			b.WriteString(s)
+		}
+		b.WriteByte('\n')
 	}
-	if err := templ.Execute(&buf, s); err != nil {
-		return nil, errSystem{err}
+	if len(rc.other) > 0 {
+		for _, o := range rc.other {
+			b.WriteString(o)
+			b.WriteByte('\n')
+		}
+	}
```

Comment:
Another silly nit ...

```suggestion
	for _, o := range rc.other {
		b.WriteString(o)
		b.WriteByte('\n')
	}
```

Reply:
Oh! Yes, redundant now; it's a remnant from an early iteration where (I think) an extra newline was conditional.

I'll update this one - I may not wait for CI to finish after that though 😅